app/script/cryptography/CryptographyRepository.js

```diff
@@ -242,46 +242,23 @@ z.cryptography.CryptographyRepository = class CryptographyRepository {
    * @returns {Promise} Resolves with the encrypted payload
    */
   encryptGenericMessage(recipients, genericMessage, payload = this._constructPayload(this.currentClient().id)) {
-    const cipherPayloadPromises = [];
-
-    for (const userId in recipients) {
-      const client_ids = recipients[userId];
-
-      payload.recipients[userId] = payload.recipients[userId] || {};
-      client_ids.forEach(client_id => {
-        const sessionId = this._constructSessionId(userId, client_id);
-        cipherPayloadPromises.push(this._encryptPayloadForSession(sessionId, genericMessage));
-      });
-    }
-
-    const receivingUsers = Object.keys(payload.recipients).length;
-    const logMessage = `Encrypting message of type '${genericMessage.content}' for '${receivingUsers}' users.`;
-    this.logger.log(logMessage, payload.recipients);
-
-    return Promise.all(cipherPayloadPromises)
-      .then(cipherPayloads => {
-        const recipientsWithMissingSessions = {};
-
-        cipherPayloads.forEach(({cipherText, sessionId}) => {
-          const {userId, clientId} = z.client.ClientEntity.dismantleUserClientId(sessionId);
-          if (!cipherText) {
-            recipientsWithMissingSessions[userId] = recipientsWithMissingSessions[userId] || [];
-            recipientsWithMissingSessions[userId].push(clientId);
-          }
-
-          payload.recipients[userId][clientId] = cipherText;
-        });
-
-        return this._encryptGenericMessageForNewSessions(recipientsWithMissingSessions, genericMessage);
+    return this._encryptGenericMessage(recipients, genericMessage, payload)
+      .then(({messagePayload, missingRecipients}) => {
+        return Object.keys(missingRecipients).length
+          ? this._encryptGenericMessageForMissingRecipients(missingRecipients, genericMessage, messagePayload)
+          : {messagePayload};
```

Comment:
I would suggest to add an empty object with the key `missingRecipients` so that it comply with the signature of the next method

Reply:
Actually, we can just send the `missingRecipients` object as part of the return.

```
{messagePayload, missingRecipients}
```

Reply:
Good point